src/strategy.py

```diff
@@ -123,6 +126,7 @@ def step(self, event_list=[]):
                         # event from the past: simulate optimal charging
                         vehicle.battery.soc = vehicle.desired_soc
                     # check that vehicle has charged enough
+                    self.margin_counter += vehicle.battery.soc < vehicle.desired_soc - self.EPS
                     if 0 <= vehicle.battery.soc < (1-self.margin)*vehicle.desired_soc - self.EPS:
```

Comment:
If we add this information to the results json, then we could delete this if statement and its content. Result would be, that the simulation doesn't break if the margin is violated (it still does if SoC becomes negative), but you get the information how often the margin was violated.

Reply:
moved counter into if-block, replaced Exception with Warning. Added final warning at end of simulation if at any time the desired SoC was not reached.